### PR TITLE
CLOUD-51844 Replace RestTemplate to Jersey client

### PR DESCRIPTION
**Pull request by mhmxs**
@akanto plz review

### CODE REVIEW

File: autoscale/src/main/java/com/sequenceiq/periscope/config/AppConfig.java [hunk 1]
```diff
@@ -63,19 +67,38 @@ public SchedulerFactoryBean schedulerFactoryBean() {
     }
 
     @Bean
-    public RestOperations createRestTemplate() throws Exception {
-        HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
-        SSLContextBuilder sslContextBuilder = new SSLContextBuilder();
-        sslContextBuilder.loadTrustMaterial(null, new TrustStrategy() {
+    public ClientBuilder jerseyClientBuilder() throws NoSuchAlgorithmException, KeyManagementException {
```
Comment (akanto) [inline autoscale/src/main/java/com/sequenceiq/periscope/config/AppConfig.java]: It looks like a copy-paste from api project

File: autoscale/src/main/java/com/sequenceiq/periscope/config/AppConfig.java [hunk 2]
```diff
@@ -63,19 +67,38 @@ public SchedulerFactoryBean schedulerFactoryBean() {
...
                 return true;
             }
-        });
-        SSLConnectionSocketFactory sslConnectionSocketFactory = new SSLConnectionSocketFactory(sslContextBuilder.build());
-        CloseableHttpClient httpClient = HttpClients.custom().setSSLSocketFactory(sslConnectionSocketFactory).build();
-        requestFactory.setHttpClient(httpClient);
-        return new RestTemplate(requestFactory);
+        };
+
+        ClientBuilder builder = ClientBuilder.newBuilder().sslContext(ctx).hostnameVerifier(hostnameVerifier)
```
Comment (akanto) [inline autoscale/src/main/java/com/sequenceiq/periscope/config/AppConfig.java]: except it is not using a proper client, like apache httpclient

File: autoscale/src/main/java/com/sequenceiq/periscope/service/security/RemoteUserDetailsService.java [hunk 3]
```diff
@@ -42,38 +44,27 @@
...
-                HttpMethod.POST,
-                new HttpEntity<>(tokenRequestHeaders),
-                Map.class).getBody();
-
-        HttpHeaders scimRequestHeaders = new HttpHeaders();
-        scimRequestHeaders.set("Authorization", "Bearer " + tokenResponse.get("access_token"));
-
-        String scimResponse;
-
+        Client client = clientBuilder.build();
```
Comment (akanto) [inline autoscale/src/main/java/com/sequenceiq/periscope/service/security/RemoteUserDetailsService.java]: Client is a heavy wight object, we shall create just a few, we already have a cache for it in api project.

File: core/src/main/java/com/sequenceiq/cloudbreak/conf/AppConfig.java [hunk 4]
```diff
@@ -171,29 +174,38 @@ public AsyncTaskExecutor containerBootstrapBuilderExecutor() {
     }
 
     @Bean
-    public RestOperations restTemplate() {
-        HttpComponentsClientHttpRequestFactory factory = new HttpComponentsClientHttpRequestFactory();
-        factory.setConnectTimeout(TIMEOUT);
+    public ClientBuilder jerseyClientBuilder() throws NoSuchAlgorithmException, KeyManagementException {
```
Comment (akanto) [inline core/src/main/java/com/sequenceiq/cloudbreak/conf/AppConfig.java]: copy paste issue, take a look at the core-api project

File: core/src/main/java/com/sequenceiq/cloudbreak/service/user/RemoteUserDetailsService.java [hunk 5]
```diff
@@ -84,39 +84,27 @@
...
-                new HttpEntity<>(tokenRequestHeaders),
-                Map.class).getBody();
-
-        HttpHeaders scimRequestHeaders = new HttpHeaders();
-        scimRequestHeaders.set("Authorization", "Bearer " + tokenResponse.get("access_token"));
-
-        String scimResponse = null;
-
+        Client client = clientBuilder.build();
+        WebTarget target = client.target(identityServerUrl).path("/oauth/token").queryParam("grant_type", "client_credentials");
```
Comment (akanto) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/user/RemoteUserDetailsService.java]: I think this one is already implemented in IdentityClient.java (or at least very similar), I would prefer  to extend that one

**Comment by doktoric** [comment]
closing this now and please create a new one when the change will be ready

**Pull request closed without merging**